server/server.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/acme/autocert"
+)
+
+func openLogFile(path string) (*os.File, error) {
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		defer f.Close()
+		return nil, err
+	}
+	return f, nil
+}
+
+type Server struct {
+	config Config
+	logger *log.Logger
+	engine *gin.Engine
+	closer func()
+}
+
+func (s *Server) Close() {
```

Comment:
Im guessing there is a better way to do this but I wanted to make sure that the sever setup was distinct from the Serve Function while not leaving anything open? I can look at this closer for us but there is probably a smarter way.

Reply:
Probably we should just collapse this into the Serve function and maybe leave New pretty basic